server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -95,7 +131,7 @@ private static boolean searchWithCollectorManager(
             queryResult.terminatedEarly(false);
         }
 
-        return topDocsFactory.shouldRescore();
+        return shouldRescore;
```

Comment:
We may need to add the interface here:

```
interface RescoringQueryCollectorContext {
    boolean shouldRescore();
}
```

```
if (queryCollectorContext instanceof RescoringQueryCollectorContext) {
    return ((RescoringQueryCollectorContext) queryCollectorContext).shouldRescore();
}
```

Reply:
makes sense, will add the interface. I assume that the default value (if the passed context is not instance of our new interface) is `false`

Reply:
Added `RescoringQueryCollectorContext` interface